client/components/theme-upgrade-modal/index.tsx

```diff
@@ -536,12 +541,41 @@ export const ThemeUpgradeModal = ( {
 		} );
 	}
 
-	const features =
-		isExternallyManaged && featureList.length === 0 ? null : (
-			<div className="theme-upgrade-modal__included">
-				<h2>{ featureListHeader }</h2>
-				<ul>
-					{ featureList.map( ( feature, i ) => (
+	// Map features list so that if list is bigger than 4, we only show 3 and "view more" link.
+	const mapFeatureList = ( featureList: any[] ) => {
+		let firstSection = featureList;
+		let secondSection = null;
+		if ( featureList.length > 4 ) {
+			firstSection = featureList.slice( 0, 3 );
+			secondSection = featureList.slice( 3 );
+		}
+		return (
```

Comment:
I wonder if this could be simpler...

```
if ( featureList.length <= 4 || isExpandedListOpen ) {
  return featureList.map(...
}

return (
  <>
    { featureList.slice( 0, 3 ).map(... }
    { View more link }
  <>
)
```

Make sense?

Reply:
Oh good point, yes, that could be simplified!